src/inmanta/agent/agent.py

```diff
@@ -372,40 +372,58 @@ class DeployRequestAction(str, enum.Enum):
     """ interrupt the current deploy: cancel it, start the new one and restart the current deploy when the new one is done"""
 
 
+# Two letter abbreviations to make table line up
+# N normal
+# P periodic
+# F full
+# I incremental
+NF = (True, False)
+NI = (False, False)
+PF = (True, True)
+PI = (False, True)
+# shorten table
+ignore = DeployRequestAction.ignore
+terminate = DeployRequestAction.terminate
+defer = DeployRequestAction.defer
+interrupt = DeployRequestAction.interrupt
+
 # This matrix describes what do when a new DeployRequest enters before the old one is done
-# Format is old_is_repair, old_is_periodic, new_is_repair, new_is_periodic
+# Format is (old_is_repair, old_is_periodic), (new_is_repair, new_is_periodic)
 # The underlying idea is that
```

Comment:
This list defines the logic for deciding which of two deploy requests gets precedence (`pick` in the branching PoC) but it doesn't define under what circumstances the other should be deferred or just skipped (`run_after` in the branching PoC).